client/src/context/reducer.ts

```diff
@@ -0,0 +1,27 @@
+import { UserDataActionType } from "./actions";
+
+export interface UserDataState {
+  uuid: string;
+}
+
+export const initialState: UserDataState = {
+  uuid:
+    typeof window !== "undefined" ? localStorage.getItem("uuid") : undefined,
```

Comment:
since you're using a transpiler, you can just use modern js and the transpiler will build down to what's specified. 

You can use optional chaining to do a nullish check. If the accessed variable or property or method is undefined OR null, it will return undefined.

```suggestion
  uuid: localStorage?.getItem?.("uuid") 
   
```